polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1029,3 +1033,125 @@ fn limit_and_sanitize_disputes<
 		(checked, checked_disputes_weight)
 	}
 }
+
+// Filters statements from disabled validators in `BackedCandidate` and `MultiDisputeStatementSet`.
+// Returns `true` if at least one statement is removed and `false` otherwise.
+fn filter_backed_statements<T: Config>(
+	backed_candidates: &mut Vec<BackedCandidate<<T as frame_system::Config>::Hash>>,
+) -> Result<bool, DispatchErrorWithPostInfo> {
+	let disabled_validators = shared::Pallet::<T>::disabled_validators();
+
+	if disabled_validators.len() == 0 {
+		// No disabled validators - nothing to do
+		return Ok(false)
+	}
+
+	// `BackedCandidate` contains `validator_indices` which are indecies within the validator group.
+	// To obtain `ValidatorIndex` from them we do the following steps:
+	// 1. Get `para_id` from `CandidateDescriptor`
+	// 2. Get the `core_idx` for the corresponding `para_id`
+	// 3. Get the validator group assigned to the corresponding core idx
+
+	// Map `para_id` to `core_idx` for step 2 from the above list
+	let para_id_to_core_idx = <scheduler::Pallet<T>>::scheduled_paras()
+		.map(|(core_idx, para_id)| (para_id, core_idx))
+		.collect::<BTreeMap<_, _>>();
```

Comment:
I see that [here](https://github.com/paritytech/polkadot-sdk/blame/21d36b7b4229c4d5225944f197918cde23fda4ea/polkadot/runtime/parachains/src/paras_inherent/mod.rs#L583-L586) we call `update_claimqueue` before `scheduled_paras`. It might be only relevant for session boundaries, but I feel uneasy duplicating the logic here and not passing this mapping like it's done in `fn process_candidates` thus introducing a chance for discrepancies in the future.

Reply:
I agree and this is fixed. The mapping is passed to this function too.